.github/workflows/nemo_v5_tests.yml

```diff
@@ -0,0 +1,222 @@
+        echo "Time-stepping duration = " $VAR_TIME
+
+    - name: NEMO 5.0 nvidia passthrough
+      run: |
+        # Set up environment
+        source /apps/spack/psyclone-spack/spack-repo/share/spack/setup-env.sh
+        spack unload && spack load nemo-build-environment%nvhpc@24.5
+        source .runner_venv/bin/activate
+        export PSYCLONE_NEMO_DIR=${GITHUB_WORKSPACE}/examples/nemo/scripts
+        export PSYCLONE_HOME=${PWD}/.runner_venv
+        export NEMO_DIR=${HOME}/NEMOv5
+        
+        # Set up FCM: PATHs are loaded from SPACK, we only need to set the FCFLAGS
+        # We compile at -O1 to permit comparison of the results.
+        cd $NEMO_DIR
+        cp $PSYCLONE_NEMO_DIR/KGOs/arch-linux_spack.fcm arch/arch-linux_spack.fcm
+        export FCFLAGS="-i4 -Mr8 -O1 -Minline -Mcray=pointer -Mpre -g"
+        
+        # Clean up and compile
+        # Without key_mpi_off it fails to compile (even without psyclone)
+        ./makenemo -r BENCH -m linux_spack -n BENCH_PASSTHROUGH_NVHPC clean -y
+        ./makenemo -r BENCH -m linux_spack -n BENCH_PASSTHROUGH_NVHPC -p ${PSYCLONE_NEMO_DIR}/passthrough.py \
+          del_key "key_xios key_iomput key_top" add_key "key_mpi_off key_nosignedzero" -j 4 -v 1
+
+        # Run Orca2
+        cd $NEMO_DIR/tests/BENCH_PASSTHROUGH_NVHPC/EXP00
+        cp $PSYCLONE_NEMO_DIR/KGOs/namelist_cfg_orca2_like namelist_cfg
+        mpirun -np 4 ./nemo
+        python $PSYCLONE_NEMO_DIR/compare_ouput.py $PSYCLONE_NEMO_DIR/KGOs/run.stat.bench.orca2.4mpi.10steps run.stat
+        cat run.stat
+        export VAR_TIME=$(awk '/ step /{print $3}' timing.output | head -n 1 | sed -e 's/s//')
+        echo "Time-stepping duration = " $VAR_TIME
+
+    - name: NEMO 5.0 Intel passthrough
+      run: |
+        # Set up environment
+        source /apps/spack/psyclone-spack/spack-repo/share/spack/setup-env.sh
+        spack unload && spack load nemo-build-environment%oneapi
+        source .runner_venv/bin/activate
+        export PSYCLONE_NEMO_DIR=${GITHUB_WORKSPACE}/examples/nemo/scripts
+        export PSYCLONE_HOME=${PWD}/.runner_venv
+        export NEMO_DIR=${HOME}/NEMOv5
+        
+        # Set up FCM: PATHs are loaded from SPACK, we only need to set the FCFLAGS
+        cd $NEMO_DIR
+        cp $PSYCLONE_NEMO_DIR/KGOs/arch-linux_spack.fcm arch/arch-linux_spack.fcm
+        export FCFLAGS="-i4 -r8 -O2 -fp-model precise -fno-alias -g"
+        
+        # Clean up and compile
+        ./makenemo -r BENCH -m linux_spack -n BENCH_PASSTHROUGH_ONEAPI clean -y
+        ./makenemo -r BENCH -m linux_spack -n BENCH_PASSTHROUGH_ONEAPI -p ${PSYCLONE_NEMO_DIR}/passthrough.py \
+          del_key "key_xios key_iomput key_top" add_key "key_nosignedzero" -j 4 -v 1
+
+        # Run Orca2
+        cd $NEMO_DIR/tests/BENCH_PASSTHROUGH_ONEAPI/EXP00
+        cp $PSYCLONE_NEMO_DIR/KGOs/namelist_cfg_orca2_like namelist_cfg
+        # It currently segfaults
```

Comment:
This script file seems to be doing nvfortran? but ESMWF uses intel and I contacted them to use the exact same steps.

Regardless, the same issue also happens when building without psyclone (which now I added to the comment) in my laptop and glados, so I will blame our spack software stack recipe. This is quite time-consuming to debug and rebuild so I also propose leaving it for a follow up, but I still wanted to include the part that works as this may already flag some psyclone issues.

Reply:
Aha, it was a small problem in the spack recipe with the cleanup (unloading of previous envvars) which is now fixed, and now it runs and generates the right results.